packages/sdk/src/wallet/DefaultSmartWallet.ts

```diff
@@ -161,26 +156,41 @@ export class DefaultSmartWallet extends SmartWallet {
    */
   async lendExecute(
     amount: number,
-    asset: AssetIdentifier,
+    asset: Asset,
     chainId: SupportedChainId,
     marketId?: string,
     options?: LendOptions,
   ): Promise<LendTransaction> {
-    // Parse human-readable inputs
-    const { amount: parsedAmount, asset: resolvedAsset } = parseLendParams(
-      amount,
-      asset,
-      chainId,
-    )
+    // Validate amount
+    if (amount <= 0) {
+      throw new Error('Amount must be greater than 0')
+    }
+
+    // Check if asset is supported on the chain
+    const tokenAddress = asset.address[chainId]
+    if (!tokenAddress) {
+      throw new Error(
+        `${asset.metadata.symbol} not supported on chain ${chainId}`,
+      )
+    }
```

Comment:
This whole function will change later, just wanted to make sure it works with the new Asset shape for now